lib/rake/test.rake

```diff
@@ -48,12 +48,18 @@ namespace :test do
         message = 'ಠ_ಠ Eyes tests for <b>dashboard</b> failed. See <a href="https://eyes.applitools.com/app/sessions/">the console</a> for results or to modify baselines.'
         ChatClient.log message, color: 'red'
         ChatClient.message 'server operations', message, color: 'red', notify: 1
+        raise "Eyes tests failed"
       end
     end
   end
 
-  # do the eyes and sauce labs ui tests in parallel
-  multitask ui_all: [:eyes_ui, :regular_ui]
+  # Run the eyes tests and ui test suites in parallel. If one of these suites
+  # raises, allow the other suite to complete, then make sure this task raises.
+  task :ui_all do
+    Parallel.each([:eyes_ui, :regular_ui], in_threads: 2) do |target|
+      RakeUtils.rake_stream_output "test:#{target}"
+    end
+  end
```

Comment:
Surprising that `multitask` doesn't already allow the other ongoing tasks to complete (or allow this to be configured at least). My tests confirm that all tasks abort on the first raised error so this looks like the only way to achieve what we want here.

However, we might be able to use `Rake::Task["test:#{target}"].invoke` rather than `RakeUtils.rake` to avoid the extra-process overhead.

Reply:
Thanks @wjordan , `Rake::Task[...].invoke` appears to have worked.